scripts/sars_cov_2/generate_pipeline_results_files.py

```diff
@@ -623,25 +657,30 @@ def generate_pipeline_results_files(
         EXPECTED_PRIMER_AUTODETECTION_HEADERS,
         PRIMER_AUTODETECTION_SAMPLE_ID_COL,
     )
-    df_ncov = load_data_from_csv(ncov_qc_csv_file, EXPECTED_NCOV_HEADERS, NCOV_SAMPLE_ID_COL)
+    df_ncov_qc = load_data_from_csv(ncov_qc_csv_file, EXPECTED_NCOV_HEADERS, NCOV_SAMPLE_ID_COL)
+    df_ncov_typing = load_data_from_csv(ncov_typing_csv_file, EXPECTED_TYPING_HEADERS, TYPING_SAMPLE_ID_COL)
     df_pangolin = load_data_from_csv(pangolin_csv_file, EXPECTED_PANGOLIN_HEADERS, PANGOLIN_SAMPLE_ID_COL)
 
     all_samples = df_metadata[SAMPLE_ID].tolist()
 
+    # typing runs as part of ncov, so they share the notifications
     qc_unrelated_failing_samples, events = _generate_notifications(
-        analysis_run_name, all_samples, df_contamination_removal, df_primer_autodetection, df_ncov, df_pangolin
+        analysis_run_name, all_samples, df_contamination_removal, df_primer_autodetection, df_ncov_qc, df_pangolin
     )
 
     _generate_results_csv(
         all_samples,
         df_contamination_removal,
         df_primer_autodetection,
-        df_ncov,
+        df_ncov_qc,
+        df_ncov_typing,
         df_pangolin,
         qc_unrelated_failing_samples,
-        output_csv_file,
+        output_results_csv_file,
     )
 
+    csv_to_json(output_results_csv_file, output_results_json_file, SAMPLE_ID)
```

Comment:
ideally this would be called by every pipeline

Reply:
That's correct. The code is in the utility dir.